hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/viewfs/TestViewFsOverloadSchemeLocalFileSystem.java

```diff
@@ -0,0 +1,158 @@
+ * file://mountTableName/ i.e, the authority is used to load a mount table.
+ */
+public class TestViewFsOverloadSchemeLocalFileSystem {
+  private static final String FILE = "file";
+  private static final Log LOG =
+      LogFactory.getLog(TestViewFsOverloadSchemeLocalFileSystem.class);
+  private FileSystem fsTarget;
+  private Configuration conf;
+  private Path targetTestRoot;
+  private FileSystemTestHelper fileSystemTestHelper;
+
+  @Before
+  public void setUp() throws Exception {
+    conf = new Configuration();
+    conf.set(String.format("fs.%s.impl",
+        FILE),
+        ViewFsOverloadScheme.class.getName());
+    conf.set(String.format(
+        FsConstants.FS_VIEWFS_OVERLOAD_SCHEME_TARGET_FS_IMPL_PATTERN,
+        FILE),
+        LocalFileSystem.class.getName());
+    fsTarget = new LocalFileSystem();
+    fsTarget.initialize(new URI("file:///"), conf);
+    fileSystemTestHelper = new FileSystemTestHelper();
+    // create the test root on local_fs
+    targetTestRoot = fileSystemTestHelper.getAbsoluteTestRootPath(fsTarget);
+    fsTarget.delete(targetTestRoot, true);
+    fsTarget.mkdirs(targetTestRoot);
+  }
+
+  /**
+   * Tests write file and read file with ViewFSOverloadScheme.
+   */
+  @Test
+  public void testLocalTargetLinkWriteSimple() throws IOException {
+    LOG.info("Starting testLocalTargetLinkWriteSimple");
+    final String testString = "Hello Local!...";
+    final Path lfsRoot = new Path("/lfsRoot");
+    ConfigUtil.addLink(conf, lfsRoot.toString(),
+        URI.create(targetTestRoot + "/local"));
+    final FileSystem lViewFs = FileSystem.get(URI.create("file:///"), conf);
+
+    final Path testPath = new Path(lfsRoot, "test.txt");
+    final FSDataOutputStream fsDos = lViewFs.create(testPath);
+    try {
+      fsDos.writeUTF(testString);
+    } finally {
+      fsDos.close();
+    }
+
+    FSDataInputStream lViewIs = lViewFs.open(testPath);
+    try {
+      Assert.assertEquals(testString, lViewIs.readUTF());
+    } finally {
+      lViewIs.close();
+    }
+  }
+
+  /**
+   * Tests create file and delete file with ViewFSOverloadScheme.
+   */
+  @Test
+  public void testLocalFsCreateAndDelete() throws Exception {
+    LOG.info("Starting testLocalFsCreateAndDelete");
+    ConfigUtil.addLink(conf, "mt", "/lfsroot",
+        URI.create(targetTestRoot + "/wd2"));
+    final URI mountURI = URI.create("file://mt/");
+    final FileSystem lViewFS = FileSystem.get(mountURI, conf);
+    try {
+      Path testPath = new Path(mountURI.toString() + "/lfsroot/test");
+      lViewFS.create(testPath);
+      Assert.assertTrue(lViewFS.exists(testPath));
+      lViewFS.delete(testPath, true);
+      Assert.assertFalse(lViewFS.exists(testPath));
+    } finally {
+      lViewFS.close();
+    }
+  }
+
+  /**
+   * Tests root level file with linkMergeSlash with ViewFSOverloadScheme.
+   */
+  @Test
+  public void testLocalFsLinkSlashMerge() throws Exception {
+    LOG.info("Starting testLocalFSCreateAndDelete");
```

Comment:
shouldn't this message be testLocalFsLinkSlashMerge? similar for the other test below

Reply:
thanks. I will change them in next patch.